src/main/java/com/google/api/codegen/transformer/ruby/GeneratedLibraryViewModel.java

```diff
@@ -0,0 +1,55 @@
+/* Copyright 2016 Google Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.transformer.ruby;
+
+import com.google.api.codegen.transformer.ModelTypeTable;
+import com.google.api.codegen.transformer.SurfaceNamer;
+import com.google.api.tools.framework.model.Interface;
+import com.google.common.base.Preconditions;
+
+/**
+ * ViewModel for a generated library. This is used by Ruby for the moment, but there's nothing
+ * Ruby-specific about it.
+ */
+// TODO(jcanizales): Move it to/merge it with something common to all languages. Reuse it for Ruby
+// MVVM.
+public class GeneratedLibraryViewModel {
+  private final Interface service;
+  private final SurfaceNamer namer;
+  private final ModelTypeTable typeTable;
+
+  public GeneratedLibraryViewModel(
```

Comment:
> view models are only data containers

The instances are immutable: what good does it make to calculate the values of the accessors on construction, vs on access?

Making the view model class an `AutoValue` has the following drawbacks:
* All values have to be calculated on construction, making that code very complex and hard to share between languages.
* No value can be `null`, even for things that a language doesn't use (e.g. https://github.com/googleapis/toolkit/pull/793#discussion_r91332509 ).
* The class definition ends up with 2x boilerplate, to get a `Builder` for it.
* If one language or feature needs a new field, transformers of all languages have to populate it or the constructor throws.
* The class tends to end up as a giant bag of undocumented fields. With a "regular" class, the implementation tells what an accessor returns, with an `AutoValue`, you have to fish for the places where it's constructed.
* The result is very un-OOP-like.

I tried that approach first for this PR and was burned very quickly. I'm not against `AutoValue` in general. It's very useful for data structures in which having structural equality (instead of referential equality) is necessary. But we don't need that for view models as far as I've seen, and it becomes a big pain.